ui/src/hooks/useKubernetes.ts

```diff
@@ -742,6 +781,27 @@ export const useKubernetes = (t: any) => {
 
   const ws = createConnect(t);
 
+  window.addEventListener('message', (event: MessageEvent) => {
+    const message = event.data;
+
+    switch (message.name) {
+      case 'PING': {
+        lunaId.value = message.id;
+        origin.value = event.origin;
+
+        sendEventToLuna('PONG', '', lunaId.value, origin.value);
+        break;
+      }
+      case 'PONG': {
+        lunaId.value = message.id;
+        origin.value = event.origin;
+
+        clearInterval(guaranteeInterval.value!);
+        break;
+      }
+    }
+  })
+
   if (ws) {
     socket = ws;
     socket!.binaryType = 'arraybuffer';
```

Comment:
I am sorry but it seems like there is no code provided as input in this prompt. Please upload or paste your code between curly brackets '{}'. I will then take a look at the code differences and suggest any improvements based on context.